packages/manager/apps/telecom/src/components/telecom/telephony/group/number/feature/ovhPabx/dialplan/extension/rule/rule.controller.js

```diff
@@ -4,13 +4,15 @@ import indexOf from 'lodash/indexOf';
 export default class DialplanExtensionRuleCtrl {
   /* @ngInject */
   constructor(
+    $state,
     $q,
     $translate,
     atInternet,
     autoScrollOnToggle,
     TelephonyMediator,
     TucToast,
   ) {
+    this.$state = $state;
```

Comment:
Can't be injected through resolve ?

Reply:
very difficult and lot of refactoring due to legacy code.